### PR TITLE
Check all WTS OIDCs are signed in before download/export to WS

### PR DESCRIPTION
**Pull request by mcannalte**
Jira Ticket: https://ctds-planx.atlassian.net/browse/HP-522

### New Features
 - If configured, check that all external OIDCs in WTS are signed in before export to WS, direct download

### Deployment changes
- To enable this check, set `features.exportToWorkspace.verifyExternalLogins` flag to true in gitops

### CODE REVIEW

File: src/Discovery/DiscoveryActionBar.tsx [hunk 1]
```diff
@@ -31,23 +34,63 @@ interface Props {
...
+    }
+
+    const { providers } = data;
+    const unauthenticatedProviders = providers.filter(
+      (provider) => !provider.refresh_token_expiration,
+    );
+    if (unauthenticatedProviders.length) {
+      setDiscoveryActionStatusMessage({
+        title: 'Please link your account to external repositories first.',
+        message: `The ${components.appName} connects to data from multiple sources. `
```
Comment (mfshao) [inline src/Discovery/DiscoveryActionBar.tsx:64]: can import `appname` from `localconf` instead

File: src/Discovery/DiscoveryActionBar.tsx [hunk 2]
```diff
@@ -31,23 +34,63 @@ interface Props {
...
+  method: 'GET',
+}).then(
+  ({ data, status }) => {
+    if (status !== 200) {
+      return false;
+    }
+
+    const { providers } = data;
+    const unauthenticatedProviders = providers.filter(
+      (provider) => !provider.refresh_token_expiration,
```
Comment (mfshao) [inline src/Discovery/DiscoveryActionBar.tsx:59]: can't remember what is the logic on WTS side, but if user has logged in before and refresh token has expired, will the `refresh_token_expiration` presists?

File: src/Discovery/DiscoveryActionBar.tsx [hunk 3]
```diff
@@ -16,7 +16,10 @@ import {
 import FileSaver from 'file-saver';
 import { DiscoveryConfig } from './DiscoveryConfig';
 import { fetchWithCreds } from '../actions';
-import { manifestServiceApiPath, hostname, jobAPIPath } from '../localconf';
+import {
+  manifestServiceApiPath, hostname, jobAPIPath, externalLoginOptionsUrl
```
Comment (mfshao) [inline src/Discovery/DiscoveryActionBar.tsx:20]:
```suggestion
  manifestServiceApiPath, hostname, jobAPIPath, externalLoginOptionsUrl,
```

**Review by mfshao — COMMENTED** [review]
Is this deployed somewhere?

And please update `/docs/portal_config.md`

File: src/Discovery/DiscoveryActionBar.tsx (hunk 2 above)
Comment (mcannalte) [reply src/Discovery/DiscoveryActionBar.tsx:59]: I didn't think to check this before, but if a user's refresh token is expired in wts, wts will return `none` for that idp
https://github.com/uc-cdis/workspace-token-service/blob/3f58479627b2af9b2d1210b40e2126dc1c359acb/wts/blueprints/external_oidc.py#L136

File: src/Discovery/DiscoveryActionBar.tsx [hunk 4]
```diff
@@ -16,7 +16,10 @@ import {
 import FileSaver from 'file-saver';
 import { DiscoveryConfig } from './DiscoveryConfig';
 import { fetchWithCreds } from '../actions';
-import { manifestServiceApiPath, hostname, jobAPIPath } from '../localconf';
+import {
+  manifestServiceApiPath, hostname, jobAPIPath, externalLoginOptionsUrl,
+} from '../localconf';
+import { appname } from '../localconf';
```
Comment (mfshao) [inline src/Discovery/DiscoveryActionBar.tsx:22]: you can combine this with the line above

**Review by mfshao — APPROVED** [review]
tested localy and it seems to be working fine 👍

**Pull request merged**